newsroom/news_api/utils.py

```diff
@@ -47,3 +48,25 @@ def remove_internal_renditions(item):
         item['associations']['featuremedia']['renditions'] = clean_renditions
 
     return item
+
+
+def check_association_permission(item):
+    """
+    Check if any of the products that the passed image item matches are permissioned superdesk products for the
+     company
+    :param item:
+    :return:
+    """
+    if ((item.get('associations') or {}).get('featuremedia') or {}).get('products'):
+        # Extract the products that the image matched in Superdesk
+        im_products = [p.get('code') for p in
+                       ((item.get('associations') or {}).get('featuremedia') or {}).get('products')]
+
+        # Check if the one of the companies products that has a superdesk product id matches one of the
+        # image product id's
+        sd_products = [p.get('sd_product_id') for p in get_products_by_company(g.user, None, 'news_api') if
```

Comment:
Does this assume that `sd_product_id` is used by the newsroom?
What happens if `sd_product_id` is not used in any products?

Reply:
In the initial development there was a facility to allow Superdesk products to be reflected in Newsroom products by cutting and pasting the Superdesk product id into a Newsroom product, we never used it until now. The potential issue is that in Superdesk we generally don't restrict products to item types, so if we use this facility for text items we'll need to ensure all text products don't ever natch an image.

Reply:
I would say that would be an issue, forcing the use of product id's (for customers that don't want it).

This might be a good place for a system setting to turn this filter/restriction on/off

Reply:
Fair point! I'll add a setting to disable/enable the checking!